src/sources/soundsourcestem.cpp

```diff
@@ -0,0 +1,468 @@
+#include "sources/soundsourcestem.h"
+
+#include "sources/readaheadframebuffer.h"
+
+extern "C" {
+
+#include <libavutil/avutil.h>
+#if LIBAVUTIL_VERSION_INT >= AV_VERSION_INT(57, 28, 100) // FFmpeg 5.1
+#include <libavutil/channel_layout.h>
+#endif
+
+} // extern "C"
+
+#include "util/assert.h"
+#include "util/logger.h"
+#include "util/sample.h"
+
+#if !defined(VERBOSE_DEBUG_LOG)
+#define VERBOSE_DEBUG_LOG false
+#endif
+
+namespace mixxx {
+
+namespace {
+
+// STEM constants
+constexpr int kNumStreams = 5;
+constexpr int kRequiredStreamCount = kNumStreams - 1; // Stem count doesn't include the main mix
+
+const Logger kLogger("SoundSourceSTEM");
+
+} // anonymous namespace
+
+const QString SoundSourceProviderSTEM::kDisplayName = QStringLiteral("STEM with FFmpeg");
+
+QStringList SoundSourceProviderSTEM::getSupportedFileTypes() const {
+    return {"stem.mp4", "stem.m4a"};
```

Comment:
Is one with video? Or can we normalize them to only one.

Reply:
NI suggest using `stem.mp4` and provide stem samples with the `stem.mp4`, however a few tools online (such as stemgen) generate `stem.m4a` files. Traktor supports both `mp4` and `m4a` extension.

Reply:
OK